### PR TITLE
Adds a new mini megafauna

### PR DESCRIPTION
**Pull request by SmoSmoSmoSmok**
## About The Pull Request
The Exiled King:
mugshot

(yeah ITS codersprite)
a former king who once ruled a proud oceanic planet, driven to madness by power until his nation rebelled against him and exiled him to the wastes of lavaland. but as u'll find out in the fight, he still has a few supporters left.  this is meant to be an easy difficulty bossfight so even the newer players can take on this boss. 

https://user-images.githubusercontent.com/95004236/235338309-d4cebfc9-16ad-40f9-9a8c-9bf24808fbbc.mp4

the king is a coward, he prefers fighting with long-range attacks as he cowers behind his soldiers. as the fight progresses, stronger minions will be summoned. For every minion killed, the king loses some HP, the minions will despawn after a short while tho so they wont pile up and overhwhelm u. slaying the king nets you a cloak for the neck slot which provides a small all-around defense boost.

## Why It's Good For The Game
i went through the trouble of turning this fauna into a basic mob so we wouldnt have to need more tg edits

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
You should make basic mob megafauna on TG since you got the base shit done already, it'll be very appreciated im sure

File: fulp_modules/features/exclusive_fauna/exiled_king_basic.dm [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+
+	charge_ability.Activate(final_target)
+
+/datum/ai_behavior/basic_ranged_attack/finish_action(datum/ai_controller/controller, succeeded, target_key, targetting_datum_key, hiding_location_key)
+	. = ..()
+	if(!succeeded)
+		controller.blackboard -= target_key
+		return
+
+/obj/item/clothing/neck/cloak/squid
```
Comment (JohnFulpWillard) [inline fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:534]: wasn't cloak armor not allowed for some reason? I think people are able to make themselves 100% immune with it.

File: fulp_modules/features/exclusive_fauna/exiled_king_basic.dm [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+	///score of killings
+	var/score_achievement_type = /datum/award/score/king_score
+
+
+/mob/living/basic/exiled_king/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/gps, "Sunken Signal")
+	add_traits(list(TRAIT_LAVA_IMMUNE, TRAIT_ASHSTORM_IMMUNE), INNATE_TRAIT)
+	AddElement(/datum/element/simple_flying)
+	AddElement(/datum/element/crusher_loot, /obj/item/crusher_trophy/bileworm_spewlet, 15)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:55]: no unique crusher trophy?

Comment (SmoSmoSmoSmok) [reply fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:55]: ill think up of a unique crusher before i undraft this

File: fulp_modules/features/exclusive_fauna/exiled_king_basic.dm [hunk 3]
```diff
@@ -0,0 +1,702 @@
+#define BB_KING_PORTAL "BB_king_portal"
+#define BB_KING_TENTACLE "BB_king_tentacle"
+#define BB_KING_SURROUND_TENTACLE "BB_king_surround_tentacle"
+#define BB_CHARGE_ABILITY "BB_charge_ability"
+#define BOSS_MEDAL_EXILED "Exiled Killer"
+#define EXILED_KING_SCORE "Exiled Killed"
+#define BB_KING_TENTACLE_TRACK "BB_king_tentacle_track"
+
+/mob/living/basic/exiled_king
+	name = "Exiled King"
```
Comment (SgtHunk) [inline fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:10]:
```suggestion
	name = "exiled king"
```

File: fulp_modules/features/exclusive_fauna/exiled_king_basic.dm [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+	desc = "and what army?"
```
Comment (SgtHunk) [inline fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:11]:
```suggestion
	desc = "And what army?"
```
either that or make it a bit more objective like other fauna ("A monstrous creature protected by heavy shielding.")

File: fulp_modules/features/exclusive_fauna/exiled_king_basic.dm [hunk 5]
```diff
@@ -0,0 +1,702 @@
...
+	sentience_type = SENTIENCE_BOSS
+	faction = list(FACTION_MINING, FACTION_BOSS)
+	basic_mob_flags = DEL_ON_DEATH
+	move_force = MOVE_FORCE_OVERPOWERING
+	move_resist = MOVE_FORCE_OVERPOWERING
+	pull_force = MOVE_FORCE_OVERPOWERING
+	layer = LARGE_MOB_LAYER
+	plane = GAME_PLANE_UPPER_FOV_HIDDEN
+	mouse_opacity = MOUSE_OPACITY_OPAQUE
+	death_message = "the mad king is felled, no longer will he suffer. "
```
Comment (SgtHunk) [inline fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:50]: Grammar

File: fulp_modules/features/exclusive_fauna/exiled_king_basic.dm [hunk 6]
```diff
@@ -0,0 +1,702 @@
...
+			if(content.density)
+				dense_check = TRUE
+				break
+		if(dense_check)
+			continue
+		var/obj/structure/cthulu_rift/portal = new(place, number_of_mobs, creature)
+		portal.squid = WEAKREF(owner)
+
+/obj/structure/cthulu_rift
+	name = "carp rift"
```
Comment (SgtHunk) [inline fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:270]: carp rift ?

File: fulp_modules/features/exclusive_fauna/exiled_king_basic.dm [hunk 7]
```diff
@@ -0,0 +1,702 @@
...
+	maxHealth = 50
+	mob_size = MOB_SIZE_LARGE
+	///our overlord
+	var/datum/weakref/leader
+	melee_damage_lower = 5
+	speed = 5
+	melee_damage_upper = 5
+
+/mob/living/basic/carp/cthulu/mega
+	name = "Overlord's commander"
```
Comment (SgtHunk) [inline fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:321]: names shouldn't ever be uppercase

File: fulp_modules/features/exclusive_fauna/exiled_king_basic.dm [hunk 8]
```diff
@@ -0,0 +1,702 @@
...
+		var/check_existing_arm = FALSE
+		for(var/obj/effect/kraken_arm/arm in singular)
+			check_existing_arm = TRUE
+			break
+		if(!check_existing_arm)
+			new /obj/effect/kraken_arm/original(singular, direction)
+		sleep(0.175 SECONDS)
+
+/obj/effect/kraken_arm
+	name = "Kraken Arm"
```
Comment (SgtHunk) [inline fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:390]: same here

File: fulp_modules/features/exclusive_fauna/exiled_king_basic.dm [hunk 9]
```diff
@@ -0,0 +1,702 @@
...
+			break
+		if(!check_existing_arm)
+			new /obj/effect/kraken_arm(singular)
+		playsound(singular, 'sound/effects/meteorimpact.ogg', 100, TRUE)
+		sleep(0.1 SECONDS)
+
+
+/mob/living/basic/vanguard
+	name = "king's vanguard"
+	desc = "vanguard of the king's army."
```
Comment (SgtHunk) [inline fulp_modules/features/exclusive_fauna/exiled_king_basic.dm:500]:
```suggestion
	desc = "Vanguard of the king's army."
```

File: fulp_modules/mapping/ruins/lavaland/exiled_ruins/exiled_ruins.dm [hunk 10]
```diff
@@ -0,0 +1,8 @@
+/datum/map_template/ruin/lavaland/fulp/exiled
+	name = "King's Ruins"
+	id = "exiled_ruins"
+	description = "A most blatant boss arena"
```
Comment (SgtHunk) [inline fulp_modules/mapping/ruins/lavaland/exiled_ruins/exiled_ruins.dm:4]: desc is fine but should be better, it's for admins iirc

**Review by SgtHunk — CHANGES_REQUESTED** [review]
boboshkan clearly doesn't use grammarly

**Comment by SmoSmoSmoSmok** [comment]
ive additionally fixed MH items spawning out of ur bag if its full

**Comment by SmoSmoSmoSmok** [comment]
WHERE are the maintainers

**Comment by SgtHunk** [comment]
sorry i was eepy

**Pull request merged**